### PR TITLE
PAXURL-329 @multi recurses child directories in predictable order

### PR DESCRIPTION
**Pull request by dbavery**
1) Enhanced @multi annotation to return files in predictable order for
both "repositories" and "defaultRepositories" properties
2) Moved test file to "internal" package so it can access package
scoped methods for testing
3) Added PowerMock test dependency, required for accurate validation
of order methods
4) Enhanced existing tests to validate that files are skipped to
enable safe refactoring of common code in AetherBasedResolver.

### CODE REVIEW

File: pax-url-aether/src/main/java/org/ops4j/pax/url/mvn/internal/AetherBasedResolver.java [hunk 1]
```diff
@@ -16,7 +16,6 @@
  */
```
Comment (dbavery) [inline pax-url-aether/src/main/java/org/ops4j/pax/url/mvn/internal/AetherBasedResolver.java]: NOTE: This file had a lot of trailing whitespace and tabs -- my editor did replace some of those on save. I tried to minimize the amount of these unrelated changes, but some did sneak in.

**Comment by ANierbeck** [comment]
so far looks good to me, if you like you can pull it in yourself, I just added you to the team.

**Comment by dbavery** [comment]
Sounds like you are suggesting I merge it (first time doing that) :) I'll let it sit for another couple of hours just incase I misunderstood, otherwise I will merge.

Thank you.

**Comment by ANierbeck** [comment]
yes, you did get me right ;) 
this is how ops4j works, we want people to get involved easier :)

**Pull request merged**